keep/providers/signl4_provider/signl4_provider.py

```diff
@@ -0,0 +1,149 @@
+import dataclasses
```

Comment:
the recommendation for providers is to add "validate_scopes" function that is triggered when the user clicks "connect" and let them know if the connection/authentication succeeded.